ISS/views/forum.py

```diff
@@ -316,6 +350,10 @@ def latest_threads(request):
     excluded_forums = [
         fpk for fpk, include in list(effective_prefs.items()) if not include]
 
+    for thread in Thread.objects.all():
```

Comment:
You definitely don't want to be iterating over every thread for this, see comments on other sections.